### PR TITLE
Social | Update connections UI initial state for WPCOM

### PR DESCRIPTION
**Pull request by manzoorwanijk**
Extracted from #37531 to its own PR.

## Proposed changes:

* Update connections UI initial state for WPCOM

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Ensure that `define( 'JETPACK_SOCIAL_USE_ADMIN_UI_V1', true );`  is enabled or add the blog sticker to enable the feature.
* Apply this PR to your sandbox
* Goto post editor
* Confirm that connections UI works as expected

### CODE REVIEW

**Review by spsiddarthan — COMMENTED** [review]
Looks good to me, but I haven't tested it on simple sites. Fancy a test, @gmjuhasz and get it across the line? I can do it later if you don't get to it before me.

**Review by gmjuhasz — COMMENTED** [review]
While testing on Simple sites I get this Fatal for Publicize not being there

File: projects/plugins/jetpack/_inc/lib/core-api/wpcom-fields/post-fields-publicize-connections.php [hunk 1]
```diff
@@ -251,6 +251,9 @@ public function get( $post_array, $request ) { // phpcs:ignore VariableAnalysis.
 			$output_connection['id']            = (string) $connection['unique_id'];
 			$output_connection['connection_id'] = (string) $connection['id'];
 
+			$output_connection['can_disconnect'] = Publicize::can_manage_connection( $connection );
```
Comment (spsiddarthan) [inline projects/plugins/jetpack/_inc/lib/core-api/wpcom-fields/post-fields-publicize-connections.php]: Shall we change this to `current_user_can( 'edit_others_posts' ) || get_current_user_id() === (int) $connection_data['user_id'];` till the test initialization gets fixed?

Comment (spsiddarthan) [reply projects/plugins/jetpack/_inc/lib/core-api/wpcom-fields/post-fields-publicize-connections.php]: Will probably have to do the same thing in the publicize package's equivalent file as well to keep them in sync.

File: projects/packages/publicize/src/class-publicize-base.php [hunk 2]
```diff
@@ -2079,6 +2079,17 @@ public function get_dismissed_notices() {
...
+
+	/**
+	 * Whether the current user can manage a connection.
+	 *
+	 * @param array $connection_data The connection data.
+	 *
+	 * @return bool
+	 */
+	public static function can_manage_connection( $connection_data ) {
+		return current_user_can( 'edit_others_posts' ) || get_current_user_id() === (int) $connection_data['user_id'];
```
Comment (spsiddarthan) [inline projects/packages/publicize/src/class-publicize-base.php:2091]: Would we still need this method?

File: projects/plugins/jetpack/_inc/lib/core-api/wpcom-fields/post-fields-publicize-connections.php (hunk 1 above)
Comment (manzoorwanijk) [reply projects/plugins/jetpack/_inc/lib/core-api/wpcom-fields/post-fields-publicize-connections.php]: Done. Thanks

File: projects/packages/publicize/src/class-publicize-base.php (hunk 2 above)
Comment (manzoorwanijk) [reply projects/packages/publicize/src/class-publicize-base.php:2091]: Yes, we use it in `Publicize` class in `get_all_connections_for_user` method. We can figure it out in future to use that method to keep that logic at one place.

**Review by spsiddarthan — COMMENTED** [review]
Looks good to me.

Let's make sure we test it on jetpack and simple for the jetpack folder change. 

Let's test for sanity on a jetpack social as well to ensure the package changes are good.

**Review by gmjuhasz — CHANGES_REQUESTED** [review]
I can confirm that this change did not break the new post screen on Simple sites

**Review by gmjuhasz — APPROVED** [review]
Wrong button above

**Pull request merged**